src/persist-client/src/read.rs

```diff
@@ -261,9 +339,9 @@ where
 
     /// Attempt to pull out the next values of this subscription.
     ///
-    /// The returned [`ReaderEnrichedHollowBatch`] is appropriate to use with
+    /// The returned [`LeasedBatch`] is appropriate to use with
     /// `ReadHandle::fetch_batch`.
```

Comment:
looks like this needs to be BatchFetcher::fetch_batch, here and elsewhere. while we're at it, let's change this to be a rustdoc link, so we get a CI failure if it rots again